### PR TITLE
[wallet] RPC usage stats

### PR DESCRIPTION
**Pull request by rasom**
status: ready

### CODE REVIEW

**Review by cammellos — APPROVED** [review]

File: services/rpcstats/service.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// New returns a new Service.
+func New() *Service {
+	return &Service{}
+}
+
+// APIs returns a list of new APIs.
+func (s *Service) APIs() []rpc.API {
+	return []rpc.API{
+		{
+			Namespace: "status",
```
Comment (Samyoul) [inline services/rpcstats/service.go]: I think that an API called `status` in an app called **Status** should be something very central to the API. Would you consider `rpcstats` as an alternative name for the rpc stats api namespace?

File: services/wallet/service.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ type Service struct {
 	db                  *Database
 	reactor             *Reactor
 	signals             *SignalsTransmitter
-	client              *ethclient.Client
+	client              reactorClient
```
Comment (Samyoul) [inline services/wallet/service.go]: I notice that the previous implementation used a pointer to an `*ethclient.Client`, did you deliberately chose to not have a struct pointer for your reactorClient? This comment also relates to the other cases in this PR.

**Review by Samyoul — APPROVED** [review]
Great! No major issues just a question about pointers and API namespace naming

File: services/rpcstats/service.go (hunk 1 above)
Comment (rasom) [reply services/rpcstats/service.go]: Changed to `rpcstats` and so we have two methods `rpcstats_getStats` and `rpcstats_reset`.

File: services/wallet/service.go (hunk 2 above)
Comment (rasom) [reply services/wallet/service.go]: `reactorClient` was an interface, not a struct and thus pointer wouldn't make sense there as I understand, right? But it was useless there and I replaced it in the last commit with `*walletClient` which is a struct pointer.

File: services/rpcstats/stats.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+type RPCUsageStats struct {
+	total            uint
+	counterPerMethod map[string]uint
+	rw               sync.RWMutex
+}
+
+var stats *RPCUsageStats
+
+func getInstace() *RPCUsageStats {
```
Comment (vkjr) [inline services/rpcstats/stats.go]: small misprint

Comment (rasom) [reply services/rpcstats/stats.go]: danke

**Pull request merged**